budgie-hotcorners/src/budgie-hotcorners.vala

```diff
@@ -0,0 +1,442 @@
+                        fired = false;
+                    }
+                    curr_area = new_area;
+                }
+                return true;
+            });
+        }
+
+        private void get_screensaver_active () {
+            try {
+                screensaver_runs = screensaverclient.GetActive();
+            }
+            catch (Error e) {
+                stderr.printf ("%s\n", e.message);
+                screensaver_runs = false;
+            }
+        }
+
+        private bool service_ison (string id) {
+            string[] names = get_dbus_namelist();
+            //  string hotc = "org.UbuntuBudgie.HotCornerSwitch";
+            for (int i=0; i<names.length; i++) {
+                if (id == names[i]) {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        private string[] get_dbus_namelist () {
+            try {
+                return freed_client.ListNames();
+            }
+            catch (Error e) {
+                stderr.printf ("%s\n", e.message);
+                return {};
+            }
+        }
+
+        private void get_screensaver_client () {
+            try {
+                screensaverclient = Bus.get_proxy_sync (
+                    BusType.SESSION, "org.gnome.ScreenSaver",
+                    ("/org/gnome/ScreenSaver")
+                );
+            }
+            catch (Error e) {
+                stderr.printf ("%s\n", e.message);
+            }
+        }
+
+        private bool get_mousedown () {
+            if (!service_ison("org.UbuntuBudgie.ShufflerInfoDaemon")) {
+                return false;
+            }
+            try {
+                return shufflerinfoclient.GetMouseIsdown(1);
+            }
+            catch (Error e) {
+                stderr.printf ("%s\n", e.message);
+            }
+            return false;
+        }
+
+        private void setup_freed_client () {
+            try {
+                freed_client = Bus.get_proxy_sync (
+                    BusType.SESSION, "org.freedesktop.DBus",
+                    ("/org/freedesktop/DBus")
+                );
+            }
+            catch (Error e) {
+                stderr.printf ("%s\n", e.message);
+            }
+        }
+
+        private void get_shuffler_client () {
+            try {
+                shufflerinfoclient = Bus.get_proxy_sync (
+                    BusType.SESSION, "org.UbuntuBudgie.ShufflerInfoDaemon",
+                    ("/org/ubuntubudgie/shufflerinfodaemon")
+                );
+            }
+            catch (Error e) {
+                stderr.printf ("%s\n", e.message);
+            }
+        }
+
+        private void run_command (string cmd) {
+            /* execute the command */
+            try {
+                Process.spawn_command_line_async(cmd);
+            }
+            catch (GLib.SpawnError err) {
+                /*
+                * in case an error occurs, the command most likely is
```

Comment:
probably should have some sort of debug warning to capture failures

Reply:
(More than) most likely, user set an invalid command, probably should not be considered a debug action. Will make it a message.